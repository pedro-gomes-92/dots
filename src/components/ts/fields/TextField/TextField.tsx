import React, { FunctionComponent } from 'react';
import classnames from 'classnames';
import { BaseFieldTypeProps as BaseFieldProps, BaseField } from '../BaseField';
import { TextInput } from '../../inputs';

export interface TextFieldProps extends BaseFieldProps {}

export const TextField: FunctionComponent<TextFieldProps> = ({
  model,
  placeHolder,
  value,
  isReadOnly,
  onChange,
  className,
  size,
  ...rest
}: TextFieldProps): JSX.Element => (
  <BaseField className={classnames('is-text', className)} {...rest}>
    <TextInput
      model={model}
      value={value}
      onChange={onChange}
      isReadOnly={isReadOnly}
      size={size}
      placeHolder={placeHolder}
    />
  </BaseField>
);
