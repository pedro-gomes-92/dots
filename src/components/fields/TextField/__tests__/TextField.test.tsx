import React from 'react';
import { render } from '@testing-library/react';

import { TextField } from '../TextField';

describe('Given <TextField />', (): void => {
  describe('When initializes', (): void => {
    it('Then should match the snapshot', (): void => {
      const { container } = render(<TextField />);
      expect(container).toMatchSnapshot();
    });
  });
});
