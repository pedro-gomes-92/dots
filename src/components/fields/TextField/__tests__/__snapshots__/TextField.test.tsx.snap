// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <TextField /> When initializes Then should match the snapshot 1`] = `
<div>
  <div
    class="field is-text"
  >
    <div
      class="field-body"
    >
      <div
        class="control"
      >
        <input
          class="input is-text"
          type="text"
        />
      </div>
    </div>
  </div>
</div>
`;
